MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -192,6 +192,8 @@ extension MMEEventsManager {
     func navigationArriveEvent(routeController: RouteController) -> [String: Any] {
         var eventDictionary = self.addDefaultEvents(routeController: routeController)
         eventDictionary["event"] = MMEEventTypeNavigationArrive
+        eventDictionary["totalDestinationWaypoints"] = routeController.routeProgress.route.legs.count
+        eventDictionary["remainingDestinationWaypoints"] = routeController.routeProgress.remainingWaypoints.count
```

Comment:
@ericrwolfe are you down with these key additions and their implications?

Reply:
Yeah, makes sense, could you also be sure to update the spec + increment the events version if you add/change keys?

Also, would it make sense to add these to all the events? Or rename to `currentLegIndex`, `currentStepIndex`, `totalLegCount`, `totalStepCount`?

Reply:
Yeah, let's make these changes.